tests/core/training/story_reader/test_common_story_reader.py

```diff
@@ -0,0 +1,258 @@
+        use_story_concatenation=False,
+        tracker_limit=1000,
+        remove_duplicates=False,
+    )
+    assert len(trackers) == 7
+    # this should be the story simple_story_with_only_end -> show_it_all
+    # the generated stories are in a non stable order - therefore we need to
+    # do some trickery to find the one we want to test
+    tracker = [t for t in trackers if len(t.events) == 5][0]
+    assert tracker.events[0] == ActionExecuted("action_listen")
+    assert tracker.events[1] == UserUttered(
+        "simple",
+        intent={"name": "simple", "confidence": 1.0},
+        parse_data={
+            "text": "/simple",
+            "intent_ranking": [{"confidence": 1.0, "name": "simple"}],
+            "intent": {"confidence": 1.0, "name": "simple"},
+            "entities": [],
+        },
+    )
+    assert tracker.events[2] == ActionExecuted("utter_default")
+    assert tracker.events[3] == ActionExecuted("utter_greet")
+    assert tracker.events[4] == ActionExecuted("action_listen")
+
+
+@pytest.mark.parametrize(
+    "stories_file",
+    [
+        "data/test_stories/stories_checkpoint_after_or.md",
+        "data/test_yaml_stories/stories_checkpoint_after_or.yml",
+    ],
+)
+async def test_can_read_test_story_with_checkpoint_after_or(
+    stories_file: Text, default_domain
+):
+    trackers = await training.load_data(
+        stories_file,
+        default_domain,
+        use_story_concatenation=False,
+        tracker_limit=1000,
+        remove_duplicates=False,
+    )
+    # there should be only 2 trackers
+    assert len(trackers) == 2
+
+
+@pytest.mark.parametrize(
+    "stories_file",
+    [
+        "data/test_stories/stories_with_cycle.md",
+        "data/test_yaml_stories/stories_with_cycle.yml",
+    ],
+)
+async def test_read_story_file_with_cycles(stories_file: Text, tmpdir, default_domain):
+    graph = await training.extract_story_graph(stories_file, default_domain)
+
+    assert len(graph.story_steps) == 5
+
+    graph_without_cycles = graph.with_cycles_removed()
+
+    assert graph.cyclic_edge_ids != set()
+    # sorting removed_edges converting set converting it to list
+    assert graph_without_cycles.cyclic_edge_ids == list()
+
+    assert len(graph.story_steps) == len(graph_without_cycles.story_steps) == 5
+
+    assert len(graph_without_cycles.story_end_checkpoints) == 2
+
+
+@pytest.mark.parametrize(
+    "stories_file",
+    [
+        "data/test_stories/stories_with_cycle.md",
+        "data/test_yaml_stories/stories_with_cycle.yml",
+    ],
+)
+async def test_generate_training_data_with_cycles(stories_file: Text, default_domain):
+    featurizer = MaxHistoryTrackerFeaturizer(
+        BinarySingleStateFeaturizer(), max_history=4
+    )
+    training_trackers = await training.load_data(
+        stories_file, default_domain, augmentation_factor=0
+    )
+    training_data = featurizer.featurize_trackers(training_trackers, default_domain)
+    y = training_data.y.argmax(axis=-1)
```

Comment:
what's the point of checking the features here?

Reply:
same as above - I didn't go too deep into existing tests.